### PR TITLE
Image Edit Step 4: Add image edit functionalities to ImageEdit plugin

### PR DESCRIPTION
**Pull request by JiuqingSong**
(For a full code change, see here: https://github.com/microsoft/roosterjs/pull/572)

This is the 4th step of ImageEdit. It add the image edit functionalities to ImageEdit plugin so that we can really use this plugin to edit image.

### CODE REVIEW

**Review by juliaroldi — APPROVED** [review]

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Resizer.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            : Math.max(base.width + deltaX * (x == 'w' ? -1 : 1), options.minWidth);
+        let newHeight = verticalOnly
+            ? base.height
+            : Math.max(base.height + deltaY * (y == 'n' ? -1 : 1), options.minHeight);
+
+        if (shouldPreserveRatio && ratio > 0) {
+            newHeight = Math.min(newHeight, newWidth / ratio);
+            newWidth = Math.min(newWidth, newHeight * ratio);
+
+            if (ratio > 0) {
```
Comment (flyingbee2012) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Resizer.ts:37]: no need to check again?

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Cropper.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        .map(layer => [0, 1].map(dir => getCropHandleHTMLInternal(layer, dir)).join(''))
+        .join('');
+}
+
+function getCropHandleHTMLInternal(layer: number, dir: number): string {
+    const position =
+        dir == 0
+            ? `right:${layer}px;height:${CROP_HANDLE_WIDTH - layer * 2}px;`
+            : `top:${layer}px;width:${CROP_HANDLE_WIDTH - layer * 2}px;`;
+    const bgColor = layer == 0 ? 'white' : 'black';
```
Comment (flyingbee2012) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Cropper.ts:105]: it should be working under dark mode, right?

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Cropper.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ */
+const Cropper: DragAndDropHandler<DragAndDropContext, CropInfo> = {
+    onDragStart: ({ editInfo }) => ({ ...editInfo }),
+    onDragging: ({ editInfo, x, y, options }, e, base, dx, dy) => {
+        [dx, dy] = rotateCoordinate(dx, dy, editInfo.angle);
+
+        const { width, height, left, right, top, bottom } = editInfo;
+        const { minWidth, minHeight } = options;
+
+        const fullWidth = width / (1 - left - right);
```
Comment (Lufedi) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Cropper.ts:29]: is it possible a division by 0 here?

**Review by Lufedi — APPROVED** [review]

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Cropper.ts (hunk 2 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/imageEditors/Cropper.ts:120]: yes, this is always on top of image, so color mode won't impact

**Review by flyingbee2012 — APPROVED** [review]

**Pull request merged**